Intel rework

milljm opened the pull request.

Commit:
Re-work Intel stack, add optimizations

check space requirements
Need to know the Mac integer version
fix llvm bootstrapping when it does not need to
Revert back to intel 2017
Allow package builder to build with no specific intel version set

Commit:
Re-work how our Intel stack is generated

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/219596/) on b041b62 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/157/site)

This comment will be updated on new commits.

Commit:
Separate Intel from the GNU stack

Any thing Intel related will now be in its own root directory.
Adjusting MODULEFILES will be used for the separation process.

Do not build portions of the Intel stack with our GNU GCC stack.
There are known issues with mixing and matching certain versions
of the two. So far, known issues that this PR ran into:

Intel 2017 does not support GCC 7.x
https://software.intel.com/en-us/forums/intel-c-compiler/topic/742701

Intel 2018 with latest versions of glibc:
https://software.intel.com/en-us/forums/intel-c-compiler/topic/777003
(Ubuntu 18, Fedora 28, OpenSUSE 15.0 are not supported)
